### PR TITLE
test: Move linter implementation into `test/linter/` subdirectory

### PR DESCRIPTION
**Pull request by ExE-Boss**
See #4688 for details.

### CODE REVIEW

File: test/lint.js [hunk 1]
```diff
@@ -77,7 +79,7 @@ function load(...files) {
 
         try {
           if (file.indexOf('browsers' + path.sep) !== -1) {
-            hasSchemaErrors = testSchema(file, './../schemas/browsers.schema.json');
+            hasSchemaErrors = testSchema(file, './../../schemas/browsers.schema.json');
```
Comment (Elchi3) [inline test/lint.js]: In line 150 in this file, the same path needs to change.

File: test/linter/test-versions.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ function checkVersions(supportData, relPath, logger) {
  * @param {string} filename
  */
 function testVersions(filename) {
-  const relativePath = path.relative(path.resolve(__dirname, '..'), filename);
+  const relativePath = path.relative(path.resolve(__dirname, '..', '..'), filename);
```
Comment (Elchi3) [inline test/linter/test-versions.js]: `relativePath` is never used? Maybe I'm overseeing something or this is a new (but separate) issue.

Comment (ExE-Boss) [reply test/linter/test-versions.js]: Yeah, it’s not being used anymore.

**Comment by Elchi3** [comment]
Thanks, @ExE-Boss this looks good to me now!
I had to merge the feature sorting enforcement PR first, so new PRs will not come in unsorted anymore. So, unfortunately you'll have to rebase this PR now. Sorry about that.

**Pull request merged**